### PR TITLE
Do not add resource limit for mem

### PR DESCRIPTION
**Pull request by kathryn-zhou**
## Changes proposed in this PR
- Do not add resource limit for memory

## Why are we making these changes?
- Decrease the number of OOM failures

### CODE REVIEW

File: integration/tests/cook/test_basic.py [hunk 1]
```diff
@@ -3446,9 +3446,7 @@ def test_container_initialization_timed_out(self):
     @unittest.skipUnless(util.docker_tests_enabled(), "Requires we're in an environment that requires docker images.")
     def test_memory_multiplier(self):
         mem = 64
-        command = f'mem_limit_bytes=$(cat /sys/fs/cgroup/memory/memory.limit_in_bytes) ; ' \
-                  f'echo "mem_limit_bytes = $mem_limit_bytes" ; ' \
-                  f'[ $mem_limit_bytes -eq {mem*1024*1024} ]'
+        command = f'bash -c \'if [[ "${{COOK_MEMORY_REQUEST_BYTES}}" -eq {mem * 1024 * 1024} ]]; then exit 0; else exit 1; fi\''
```
Comment (kathryn-zhou) [inline integration/tests/cook/test_basic.py:3449]: not sure if this is the correct way to access env variables on the pod

File: scheduler/src/cook/kubernetes/api.clj [hunk 2]
```diff
@@ -812,10 +814,12 @@
 (defn set-mem-cpu-resources
   "Given a resources object and a CPU and memory request and limit, update the resources object to reflect the
   desired requests and limits."
-  [^V1ResourceRequirements resources memory-request memory-limit cpu-request cpu-limit]
+  [^V1ResourceRequirements resources memory-request cpu-request cpu-limit & {:keys [memory-limit] :or {memory-limit nil}}]
```
Comment (nsinkov) [inline scheduler/src/cook/kubernetes/api.clj:817]: I think it's better to revert to the old function signature. see below for what to pass in

File: scheduler/src/cook/kubernetes/api.clj [hunk 3]
```diff
@@ -905,8 +909,8 @@
                    ;; preserving compatibility with Meosos + Cook Executor.
                    (not progress-file-path)
                    (assoc progress-file-var (str workdir "/" task-id ".progress")))
-        main-env-vars (make-filtered-env-vars main-env)
-        computed-mem (if checkpoint-memory-overhead (add-as-decimals mem checkpoint-memory-overhead) mem)]
+        computed-mem (if checkpoint-memory-overhead (add-as-decimals mem checkpoint-memory-overhead) mem)
+        main-env-vars (make-filtered-env-vars main-env computed-mem)]
```
Comment (nsinkov) [inline scheduler/src/cook/kubernetes/api.clj:913]: don't change the make-filtered-env-vars function, just move `computed-mem (if ...` above `main-env (cond-> main-env-base` and add computed-mem to main-env

File: scheduler/src/cook/kubernetes/api.clj [hunk 4]
```diff
@@ -936,7 +940,10 @@
     (.setTty container true)
     (.setStdin container true)
 
-    (set-mem-cpu-resources resources computed-mem computed-mem cpus cpus)
+    ; add memory limit if config flag set-memory-limit? is True
+    (if (get (config/kubernetes) :set-memory-limit? true)
```
Comment (nsinkov) [inline scheduler/src/cook/kubernetes/api.clj:944]: change to `(set-mem-cpu-resources resources computed-mem (when (:set-memory-limit? (config/kubernetes)) computed-mem) cpus cpus)`

File: scheduler/src/cook/kubernetes/api.clj [hunk 5]
```diff
@@ -971,6 +978,7 @@
                                                              (get-in sidecar [:resource-requirements fieldname])
                                                              0))
               total-memory-request (add-as-decimals computed-mem (get-resource-requirements-fn :memory-request))
+              ; TODO: wrap in config var
```
Comment (nsinkov) [inline scheduler/src/cook/kubernetes/api.clj:981]: revert this. we actually always want limit for init container and sidecar

File: scheduler/src/cook/kubernetes/api.clj [hunk 6]
```diff
@@ -984,7 +992,9 @@
           (.setVolumeMounts container (filterv some? (concat [(init-container-workdir-volume-mount-fn false)
                                                               (scratch-space-volume-mount-fn false)]
                                                              init-container-checkpoint-volume-mounts)))
-          (set-mem-cpu-resources resources total-memory-request total-memory-limit total-cpu-request total-cpu-limit)
+          (if (get (config/kubernetes) :set-memory-limit? true)
```
Comment (nsinkov) [inline scheduler/src/cook/kubernetes/api.clj:995]: revert this. we actually always want limit for init container and sidecar

File: scheduler/src/cook/kubernetes/api.clj [hunk 7]
```diff
@@ -1016,7 +1026,9 @@
             (.setReadinessProbe container readiness-probe)))
 
           ; resources
-          (set-mem-cpu-resources resources memory-request memory-limit cpu-request cpu-limit)
+          (if (get (config/kubernetes) :set-memory-limit? true)
```
Comment (nsinkov) [inline scheduler/src/cook/kubernetes/api.clj:1029]: revert this. we actually always want limit for init container and sidecar

File: scheduler/src/cook/kubernetes/api.clj (hunk 4 above)
Comment (kathryn-zhou) [reply scheduler/src/cook/kubernetes/api.clj:944]: done

File: scheduler/src/cook/kubernetes/api.clj (hunk 5 above)
Comment (kathryn-zhou) [reply scheduler/src/cook/kubernetes/api.clj:981]: done

File: scheduler/src/cook/kubernetes/api.clj (hunk 6 above)
Comment (kathryn-zhou) [reply scheduler/src/cook/kubernetes/api.clj:995]: done

File: scheduler/src/cook/kubernetes/api.clj (hunk 7 above)
Comment (kathryn-zhou) [reply scheduler/src/cook/kubernetes/api.clj:1029]: done

File: scheduler/src/cook/kubernetes/api.clj (hunk 2 above)
Comment (kathryn-zhou) [reply scheduler/src/cook/kubernetes/api.clj:817]: done

File: scheduler/src/cook/kubernetes/api.clj (hunk 3 above)
Comment (kathryn-zhou) [reply scheduler/src/cook/kubernetes/api.clj]: done

**Pull request merged**